ci/derived-image-test.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This script performs the following tasks:
+#
+# 1. Logs into the rhcos-devel service account to get the image push credential
+# so we can push to the registry.ci.openshift.org/rhcos-devel namespace.
+#
+# 2. Kicks off the build-test-qemu.sh script to build and test an RHCOS image
+# with Kola tests.
+#
+# 3. Pushes the resulting image to the image registry using the Prow build ID
+# as the tag since we need to have the image in this registry so our derived
+# image build test can use it.
+#
+# 3. Kicks off the derived OS image testing binary which pulls the newly built
+# image, builds a derived image from it, applies it to an underlying OpenShift
+# cluster node, verifies that it was successfully applied, then rolls the node
+# back.
+
+set -euo pipefail
+
+# Note: The oc binary will be injected by the Prow CI process. It
+# is not present in ci/Dockerfile.
+oc login https://api.ci.l2s4.p1.openshiftapps.com:6443 --token="$(cat /service-account-token/image-pusher-service-account-token)"
+oc registry login --registry=registry.ci.openshift.org --to="$SHARED_DIR/dockercfg.json"
+
+export COSA_DIR="/tmp/cosa"
+mkdir -p "$COSA_DIR"
+
+# Run the cosa build / test
+/src/ci/build-test-qemu.sh
+
+export REGISTRY_AUTH_FILE="$SHARED_DIR/dockercfg.json"
+
+# Ensure we're in the designated cosa directory so the push-container commands work
+cd "$COSA_DIR"
+
+# Tags with the cosa build ID / arch - unique to this specific build
+cosa push-container registry.ci.openshift.org/rhcos-devel/rhel-coreos
+
+# Tag with the Prow Build ID because we don't want to overwrite our well-known
+# tags yet, but our test cluster needs the image to be pushed someplace so we
+# can ingest it. We use the BUILD_ID value because its unique to each job so
+# they won't stomp on each other if running concurrently.
+cosa push-container "registry.ci.openshift.org/rhcos-devel/rhel-coreos:$BUILD_ID"
+
+# Perform the derived OS image build tests
+export BASE_IMAGE_TAG="$BUILD_ID"
```

Comment:
Hmm, I think we should inject the full pull spec in the future - this would give us more flexibility.  (We'll hopefully be pushing the official images outside of `rhcos-devel` in the near future)

Reply:
This is consumed by the derived OS test, but I agree. I'll get a PR open to fix that.

Reply:
I updated the Golang test to search for `BASE_IMAGE_PULLSPEC` and default to `registry.ci.openshift.org/rhcos-devel/rhel-coreos:latest` if that env var isn't set.